lib/web/web.go

```diff
@@ -0,0 +1,70 @@
+package web
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type Response interface{}
+
+type ServerFunc func(w http.ResponseWriter, r *http.Request) (Response, error)
+
+func ResponseHandler(f ServerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		resp, err := f(w, r)
+		if err != nil {
+			handleError(w, err)
+			return
+		}
+		handleResponse(w, resp)
+	}
+}
+
+type webError struct {
+	Message string `json:"message"`
+}
```

Comment:
I think this can just be called `Error` and should be exported for when I start working on the SDK.